java-checks/src/main/java/org/sonar/java/checks/AssertsOnParametersOfPublicMethodCheck.java

```diff
@@ -54,29 +55,22 @@ public void visitNode(Tree tree) {
       return;
     }
     MethodTree methodTree = (MethodTree) tree;
-    if (!methodTree.symbol().isPublic()) {
+    if (!methodTree.symbol().isPublic() || methodTree.symbol().owner().isPrivate()) {
       return;
     }
-    for (VariableTree parameter : methodTree.parameters()) {
-      for (IdentifierTree paramUsage : parameter.symbol().usages()) {
-        Tree paramParent = paramUsage.identifierToken().parent();
-        while (!paramParent.equals(methodTree) && !assertReported.contains(paramParent)) {
-          if (parentType(paramParent)) {
-            break;
-          }
-          paramParent = paramParent.parent();
-        }
-      }
-    }
+    methodTree.parameters().stream().map(VariableTree::symbol).map(Symbol::usages).flatMap(List::stream)
```

Comment:
It's easier to read when all stream operations are on dedicated lines. Could you rewrite this?